src/openzaak/components/catalogi/admin/catalogus.py

```diff
@@ -52,6 +64,94 @@ class CatalogusAdmin(ListObjectActionsAdminMixin, admin.ModelAdmin):
     )
     inlines = (ZaakTypeInline, BesluitTypeInline, InformatieObjectTypeInline)
 
+    def get_urls(self):
+        urls = super().get_urls()
+        my_urls = [
+            path(
+                "import/",
+                self.admin_site.admin_view(self.import_view),
+                name="catalogi_catalogus_import",
+            )
+        ]
+        return my_urls + urls
+
+    def import_view(self, request):
+        form = CatalogusImportForm(request.POST, request.FILES)
+        context = dict(self.admin_site.each_context(request), form=form)
+        if "_import" in request.POST:
+            if form.is_valid():
+                try:
+                    call_command("import", form.cleaned_data["file"])
+                    self.message_user(
+                        request,
+                        _("Catalogus successfully imported"),
+                        level=messages.SUCCESS,
+                    )
+                    return HttpResponseRedirect(
+                        reverse("admin:catalogi_catalogus_changelist")
+                    )
+                except CommandError as exc:
+                    self.message_user(request, exc, level=messages.ERROR)
+        return TemplateResponse(request, "import_catalogus.html", context)
+
+    def get_related_objects(self, obj):
+        resources = {}
+
+        resources["Catalogus"] = [obj.pk]
+
+        # Resources with foreign keys to catalogus
+        fields = ["InformatieObjectType", "BesluitType", "ZaakType"]
+        for field in fields:
+            resources[field] = list(
+                getattr(obj, f"{field.lower()}_set").values_list("pk", flat=True)
+            )
+        resources["ZaakInformatieobjectType"] = list(
+            ZaakInformatieobjectType.objects.filter(
+                zaaktype__in=resources["ZaakType"],
+                informatieobjecttype__in=resources["InformatieObjectType"],
+            ).values_list("pk", flat=True)
+        )
+
+        # Resources with foreign keys to  ZaakType
+        fields = ["ResultaatType", "RolType", "StatusType", "Eigenschap"]
+        for field in fields:
+            model = apps.get_model("catalogi", field)
+            resources[field] = list(
+                model.objects.filter(zaaktype__in=resources["ZaakType"]).values_list(
+                    "pk", flat=True
+                )
+            )
+
+        resource_list = []
+        id_list = []
+        for resource, ids in resources.items():
+            if ids:
+                resource_list.append([resource])
+                id_list.append([",".join(str(id) for id in ids)])
+
+        return resource_list, id_list
+
+    def response_post_save_change(self, request, obj):
+        if "_export" in request.POST:
+            # Clear messages
+            storage = messages.get_messages(request)
+            for i in storage:
+                pass
```

Comment:
Can you check how @annashamray did this for creating a new version?

I can't remember this was needed there, but then again, it might just show 2 messages...heh.

Reply:
Yes, if I remove it, it will show 2 messages: the custom one I added and the original message that says that the resource was successfully edited or whatever